src/Parsers/Access/ASTExecuteAsQuery.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <Parsers/IAST.h>
+
+
+namespace DB
+{
+class ASTRolesOrUsersSet;
+
+/** EXECUTE AS <user>
+  */
```

Comment:
It may be also useful to enable switching this feature off:
```
EXECUTE AS <user> ON
...
EXECUTE AS <user> OFF
```

Simple `Execute AS <user>` could be made an alias of `EXECUTE AS <user> ON`.